src/Quasar/Mount/SparkFTP.purs

```diff
@@ -0,0 +1,100 @@
+{-
+Copyright 2017 SlamData, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Quasar.Mount.SparkFTP
+  ( Config
+  , toJSON
+  , fromJSON
+  , toString
+  , fromString
+  , module Exports
+  ) where
+
+import Prelude
+
+import Data.Argonaut (Json, decodeJson, jsonEmptyObject, (.?), (:=), (~>))
+import Data.Bifunctor (bimap, lmap)
+import Data.Either (Either(..))
+import Data.Maybe (Maybe(..), maybe)
+import Data.String (Pattern(..), joinWith, split)
+import Data.URI as URI
+import Data.URI.Path (printPath, parseURIPathAbs)
+
+import Quasar.Mount.Common (Host, extractHost)
+import Quasar.Mount.Common (Host) as Exports
+import Quasar.Types (AnyPath)
+
+import Text.Parsing.StringParser (runParser)
+
+type Config =
```

Comment:
> Both login and password are obligatory but we can keep them to be anonymous and a (anonymous user does not need password but we need to fill the blank in url, pass will be simply ignored)

Seems that we need 
```purescript
...
 , user :: Maybe String
 , password :: Maybe String 
...
``` 
here. And then if `Nothing` provide arbitrary strings in `toString` function.